apps/desktop/src-tauri/src/inference/process.rs

```diff
@@ -20,9 +20,11 @@ pub type ModelGuard = Arc<Mutex<Option<Box<dyn Model>>>>;
 pub struct InferenceThreadRequest {
   pub token_sender: Sender<Bytes>,
   pub abort_flag: Arc<RwLock<bool>>,
-
   pub model_guard: ModelGuard,
   pub completion_request: CompletionRequest,
+  pub nonstream_completion_tokens: Arc<Mutex<String>>,
```

Comment:
I think we can make this private if we use it as a trait state for the non-stream feature. Making it pub would allow others to inspect it while it's writing/locked, which could potentially deadlock the Mutex writer if we're not careful... :d